.github/workflows/ecosystem-ci-trigger.yml

```
@@ -9,7 +9,143 @@ jobs:
+              const reaction = await github.rest.reactions.createForIssueComment({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                comment_id: context.payload.comment.id, 
+                content: 'rocket',
+              })
+              return { exists: false, reaction: reaction.id }
+            }
+
+            return { exists: true, reaction: null }
+      - name: Generate Token
+        id: generate-token
+        uses: tibdex/github-app-token@v2
+        with:
+          app_id: ${{ secrets.ECOSYSTEM_CI_GITHUB_APP_ID }}
+          installation_retrieval_payload: "${{ github.repository_owner }}/vite-ecosystem-ci"
+          private_key: ${{ secrets.ECOSYSTEM_CI_GITHUB_APP_PRIVATE_KEY }}
+
+      - name: Trigger Preview Release (if Package Not Found)
+        if: steps.check-package.outputs.exists == false
+        uses: actions/github-script@v7
+        id: trigger-preview-release
+        with:
+          github-token: ${{ steps.generate-token.outputs.token }}
+          script: |
+            const prData = ${{ steps.get-pr-data.outputs.result }}
+            console.log('Package not found, triggering preview release...')
+
+            // Add label "trigger: preview" to the PR
+            await github.rest.issues.addLabels({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: prData.num,
+              labels: ['trigger: preview']
+            })
+            console.log('Added "trigger: preview" label.')
+
+      - name: Wait for Preview Release Completion (if Package Not Found)
+        if: steps.check-package.outputs.exists == false
+        uses: actions/github-script@v7
+        id: wait-preview-release
+        with:
+          script: |
+            const prData = ${{ steps.get-pr-data.outputs.result }}
+            const reaction = ${{ steps.check-package.outputs.reaction }}
+            const workflowFileName = 'preview-release.yml'
+            const workflow = await github.rest.actions.getWorkflow({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              workflow_id: workflowFileName,
+            })
+            const workflowId = workflow.data.id
+            console.log(`Waiting for workflow ID ${workflowId} to complete...`)
+
+            const maxRetries = 60 // Wait up to 10 minutes
+            const delay = 10000 // 10 seconds
+            let completed = false
+
+            for (let i = 0; i < maxRetries; i++) {
+              const runsData = await github.rest.actions.listWorkflowRuns({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                workflow_id: workflowId,
+                per_page: 100,
+                page: 1,
+              })
+
+              const runs = runsData.data.workflow_runs.filter(
+                run => run.head_sha === prData.commit
+              )
```

```suggestion
              const runsData = await github.rest.actions.listWorkflowRuns({
                owner: context.repo.owner,
                repo: context.repo.repo,
                workflow_id: workflowId,
                head_sha: prData.commit,
                per_page: 100,
                page: 1,
              })

              const runs = runsData.data.workflow_runs
```
It seems the API can filter with `head_sha`.
https://octokit.github.io/rest.js/v21/#actions-list-workflow-runs